api/server.go

```diff
@@ -6,30 +6,27 @@ package api
 import (
 	"encoding/json"
 	"fmt"
-	"net"
 	"net/http"
 	"net/http/pprof"
 	"strconv"
-	"time"
 
 	"github.com/mattermost/mattermost-load-test-ng/defaults"
 	"github.com/mattermost/mattermost-load-test-ng/loadtest"
-	"github.com/mattermost/mattermost-load-test-ng/loadtest/control"
-	"github.com/mattermost/mattermost-load-test-ng/loadtest/control/noopcontroller"
 	"github.com/mattermost/mattermost-load-test-ng/loadtest/control/simplecontroller"
 	"github.com/mattermost/mattermost-load-test-ng/loadtest/control/simulcontroller"
-	"github.com/mattermost/mattermost-load-test-ng/loadtest/store/memstore"
-	"github.com/mattermost/mattermost-load-test-ng/loadtest/user/userentity"
 	"github.com/mattermost/mattermost-load-test-ng/performance"
 
 	"github.com/gorilla/mux"
 	"github.com/mattermost/mattermost-server/v5/mlog"
 )
 
+type Initializer func(config *loadtest.Config, userOffset int, namePrefix string, controllerConfig interface{}) loadtest.NewController
```

Comment:
I'd probably change the order slightly. Have the configs first and then the options (prefix and offset) which might even become an optional structure at some point.

Reply:
A couple of minor notes about naming, since it's never easy task:

- I understand this is not a common function but I believe the name `Initializer` might suggest that to be an interface.
- As it is now, it might be slightly too generic but I also understand that we don't want to end up with a 40 chars name. Maybe we could follow the convention of adding a `uc` prefix if it has to do with a `UserController`.
- This is probably the less important, but personally I find the word `initialize` not to describe what we are doing here. To initialize something I usually need to have some kind of object already created. To me this is more like a *constructor* function.

Reply:
Maybe something to indicate that it is a wrapper over loadtest.NewController ? Like `NewControllerFn` or `NewControllerWrapper`. Are there any existing norms for higher-order functions? Not sure.